### PR TITLE
Bugfix for #1576732. Abnormal sigterm handling.

### PR DESCRIPTION
**Pull request by VTabolin**

### CODE REVIEW

File: vmware_dvs/agent/firewalls/vcenter_firewall.py [hunk 1]
```diff
@@ -141,10 +156,13 @@ def __init__(self):
         for x in xrange(10):
             self.list_queues.append(Queue())
         self.remove_queue = Queue()
-        self.fw_process = Process(target=firewall_updater_loop,
+        self.fw_process = Process(target=firewall_main,
             args=(self.list_queues, self.remove_queue))
```
Comment (ekosareva) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py]: Please start line 160 ('args=....')  under  'target=....'

File: vmware_dvs/agent/firewalls/vcenter_firewall.py [hunk 2]
```diff
@@ -31,19 +32,33 @@
...
+    signal.signal(signal.SIGTERM, dvs_firewall._handle_sigterm)
+    dvs_firewall.updater_loop()
 
-        dvs, ports = pq.get_update_tasks()
-        if dvs and ports > 0:
-            updater(dvs, ports)
-        else:
-            time.sleep(1)
+
+class dvs_firewall_updater(object):
```
Comment (ekosareva) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py]: Accordingly PEP, please, use CamelCase for class names: dvs_firewall_updater -> DVSFirewallUpdater

File: vmware_dvs/agent/firewalls/vcenter_firewall.py [hunk 3]
```diff
@@ -31,19 +32,33 @@
...
-def firewall_updater_loop(list_queues, remove_queue):
-    pq = PortQueue(list_queues, remove_queue)
-    pq.port_updater_loop()
-    while True:
-        r_ports = pq.get_remove_tasks()
-        if r_ports:
-            remover(pq, r_ports)
+def firewall_main(list_queues, remove_queue):
+    dvs_firewall = dvs_firewall_updater(list_queues, remove_queue)
+    signal.signal(signal.SIGTERM, dvs_firewall._handle_sigterm)
```
Comment (ekosareva) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py:37]: Does line 37 and staff for this really need? Why is't not enough 

```
self.fw_process.terminate()
self.fw_process.join()
```

File: vmware_dvs/agent/firewalls/vcenter_firewall.py (hunk 1 above)
Comment (VTabolin) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py]: Done

File: vmware_dvs/agent/firewalls/vcenter_firewall.py (hunk 2 above)
Comment (VTabolin) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py]: Done

File: vmware_dvs/agent/firewalls/vcenter_firewall.py (hunk 3 above)
Comment (VTabolin) [inline vmware_dvs/agent/firewalls/vcenter_firewall.py:37]: self.fw_process.terminate() will block self.fw_process.join() call
We want to finish last iteration and then terminate process

**Pull request merged**